x/upgrade/upgrade.go

```diff
@@ -0,0 +1,40 @@
+package upgrade
+
+type Schedule []Plan
+
+type Plan struct {
+	Start   int64
+	End     int64
+	Version uint64
+}
+
+// ShouldUpgradeNextHeight returns true if the network of the given chainID should
+// modify the app version in the following block. This can be used for both upgrading
+// and downgrading. This relies on social consensus to work. At least 2/3+ of the
```

Comment:
[question] how does this work for downgrading? If the app version must increase each upgrade, I guess a new appVersion could be created that mimics the behavior of a previous app version so it upgrades appVersion but downgrades behavior. Is that "and downgrading" alludes to?

Reply:
Downgrading isn't supported yet (I have some ideas I can look to implement in a later PR). 

Currently it would work as you predict that we need a new app version that mimics the behavior of a previous app version